weldx/core.py

```diff
@@ -470,55 +534,28 @@ def interp_time(
 
         Returns
         -------
-        xarray.DataArray:
+        TimeSeries :
             A data array containing the interpolated data.
 
         """
-        if isinstance(self._data, xr.DataArray):
-            if isinstance(time, pint.Quantity):
-                time = ut.to_pandas_time_index(time)
-            if not isinstance(time, pd.TimedeltaIndex):
-                raise ValueError(
-                    '"time" must be a time quantity or a "pandas.TimedeltaIndex".'
-                )
-            # constant values are also treated by this branch
-            if self._data.attrs["interpolation"] == "linear" or self.shape[0] == 1:
-                return ut.xr_interp_like(
-                    self._data,
-                    {"time": time},
-                    assume_sorted=False,
-                    broadcast_missing=False,
-                )
-
-            dax = self._data.reindex({"time": time}, method="ffill")
-            return dax.fillna(self._data[0])
-
-        # Transform time to both formats
-        if isinstance(time, pint.Quantity) and time.check(UREG.get_dimensionality("s")):
-            time_q = time
-            time_pd = ut.to_pandas_time_index(time)
-        elif isinstance(time, pd.TimedeltaIndex):
-            time_q = ut.pandas_time_delta_to_quantity(time, time_unit)
-            time_pd = time
-        else:
-            raise ValueError(
-                '"time" must be a time quantity or a "pandas.TimedeltaIndex".'
+        if self._interp_counter > 0:
+            warn(
+                "The data of the time series has already been interpolated "
+                f"{self._interp_counter} time(s)."
             )
 
-        if len(self.shape) > 1 and np.iterable(time_q):
-            while len(time_q.shape) < len(self.shape):
-                time_q = time_q[:, np.newaxis]
-
-        # evaluate expression
-        data = self._data.evaluate(**{self._time_var_name: time_q})
-        data = data.astype(float).to_reduced_units()  # float conversion before reduce!
+        if isinstance(self._data, xr.DataArray):
+            dax = self._interp_time_discrete(time)
+        else:
+            dax = self._interp_time_expression(time, time_unit)
 
-        # create data array
-        if not np.iterable(data):  # make sure quantity is not scalar value
-            data = np.expand_dims(data, 0)
+        interpolation = self.interpolation
+        if interpolation is None:
+            interpolation = "linear"
```

Comment:
I don't think we have to check for None here any longer
maybe we should add a setter for `self.interpolation` that also checks for the valid interpolation styles?

Reply:
Well, it is needed in the current version because TS based on mathematical expressions have no interpolation and creating a new TS with `None` as interpolation raises an exception. However, I think the problem lies in the init method. I will rewrite it.

A setter for the interpolation is certainly useful. Will add it.

Reply:
ah yes I forgot about the math expression, makes sense
looks good this way imo